connect/transforms/src/main/java/org/apache/kafka/connect/transforms/Cast.java

```diff
@@ -0,0 +1,363 @@
+                        return Byte.parseByte((String) value);
+                    else
+                        throw new ConnectException("Unexpected type in Cast transformation: " + value.getClass());
+                case INT16:
+                    if (value instanceof Number)
+                        return ((Number) value).shortValue();
+                    else if (value instanceof Boolean)
+                        return ((boolean) value) ? (short) 1 : (short) 0;
+                    else if (value instanceof String)
+                        return Short.parseShort((String) value);
+                    else
+                        throw new ConnectException("Unexpected type in Cast transformation: " + value.getClass());
+                case INT32:
+                    if (value instanceof Number)
+                        return ((Number) value).intValue();
+                    else if (value instanceof Boolean)
+                        return ((boolean) value) ? 1 : 0;
+                    else if (value instanceof String)
+                        return Integer.parseInt((String) value);
+                    else
+                        throw new ConnectException("Unexpected type in Cast transformation: " + value.getClass());
+                case INT64:
+                    if (value instanceof Number)
+                        return ((Number) value).longValue();
+                    else if (value instanceof Boolean)
+                        return ((boolean) value) ? (long) 1 : (long) 0;
+                    else if (value instanceof String)
+                        return Long.parseLong((String) value);
+                    else
+                        throw new ConnectException("Unexpected type in Cast transformation: " + value.getClass());
+                case FLOAT32:
+                    if (value instanceof Number)
+                        return ((Number) value).floatValue();
+                    else if (value instanceof Boolean)
+                        return ((boolean) value) ? 1.f : 0.f;
+                    else if (value instanceof String)
+                        return Float.parseFloat((String) value);
+                    else
+                        throw new ConnectException("Unexpected type in Cast transformation: " + value.getClass());
+                case FLOAT64:
+                    if (value instanceof Number)
+                        return ((Number) value).doubleValue();
+                    else if (value instanceof Boolean)
+                        return ((boolean) value) ? 1. : 0.;
+                    else if (value instanceof String)
+                        return Double.parseDouble((String) value);
+                    else
+                        throw new ConnectException("Unexpected type in Cast transformation: " + value.getClass());
+                case BOOLEAN:
+                    if (value instanceof Number)
+                        return ((Number) value).longValue() > 0L;
```

Comment:
Seems like we still return `false` on negative numbers. Did the fix get included?